iceoryx_hoofs/include/iceoryx_hoofs/cxx/scope_guard.hpp

```diff
@@ -54,7 +54,7 @@ class ScopeGuardWithVariableCapacity
     /// @param[in] initFunction callable which will be called in the constructor
     /// @param[in] cleanupFunction callable which will be called in the destructor
     ScopeGuardWithVariableCapacity(const function_ref<void()>& initFunction,
-                                   const function<void()>& cleanupFunction) noexcept;
+                                   const function<void(), CleanupCapacity>& cleanupFunction) noexcept;
```

Comment:
@mossmaurice It would be good to document the purpose of the `CleanupCapacity` better to make it better reviewable. Beside that we are good to go here.